content/blog/2018-05-08-react-native-navigation.md

```diff
@@ -0,0 +1,419 @@
+
+Notice that the home screen has a couple default labels. These are created in `App.js`. (Note: running on a Android simulator is a little more involved so we aren’t going to cover it in this post. Read more [here](https://developers.arcgis.com/android/10-2/sample-code/emulator/).)
+
+### Setting Up A Tab Navigator
+
+Unlike a traditional React components, `react-navigation` components aren’t implemented as a Component subclass; they’re created via a custom constructor function that can take in a number of different options.
+
+Let’s start by opening up our `App.js` file and removing the `<View />` component and the three `<Text />` components in `render()`. Now add the import statement at the top of the file.
+
+```
+import { createBottomTabNavigator } from 'react-navigation'
+```
+
+In order to create a `TabNavigator` component, that we can return from the `render()` method, we’ll need to call the `createBottomTabNavigator` constructor and set it to a variable. Here’s our new `App.js` file:
+
+```
+import React from 'react'
+import { createBottomTabNavigator } from 'react-navigation'
+import { StackOne } from './stacks/StackOne'
+import { StackTwo } from './stacks/StackTwo'
+import TabIcon from './TabIcon'
+import { Text, Platform, StyleSheet } from 'react-native'
+
+export default class App extends React.Component {
+  
+  render() {
+    return (
+      <HomeNavigation />
+    )
+  }
+
+}
+
+const HomeNavigation = createBottomTabNavigator(
+  {
+    TabOne: {
+      screen: StackOne,
+      navigationOptions: ({ navigation }) => ({
+      tabBarIcon: ({ focused, tintColor }) => {
+        return <TabIcon index={ 0 }
+                        isFocused={ focused }/>
+        }
+      })
+    },
+    TabTwo: {
+      screen: StackTwo,
+      navigationOptions: ({ navigation }) => ({
+      tabBarIcon: ({ focused, tintColor }) => {
+        return <TabIcon index={ 1 }
+                        isFocused={ focused }/>
+        }
+      })
+    }
+  }
+)
+```
+
+You’ll see how navigation components are a little different in how they’re created. Instead of sub-classing the `React.Component` class, we’re calling the exported constructor function `createBottomTabNavigator` and passing through options that will define what the `TabNavigator` will display.
+
+(Note: When naming components that are set to variables, be sure to begin the name with a capital letter. JSX will interpret a lowercase tag name to be an HTML tag. `<tabNavigator />` will compile to `React.createElement('tabNavigator')` where `<TabNavigator />` will compile to 
+`React.createElement(TabNavigator)` — we require the latter.) 
+
+We aren’t able to run this yet since we haven’t created `StackOne` and `StackTwo`. But that’s the next step!
+
+### Setting Up The Stack Navigators
+
+Start by running the following commands to create all the other files we’ll need for this project.
+
+```
+touch StackOne.js StackOneMain.js StackOneDetail.js StackTwo.js StackTwoMain.js StackTwoDetail.js
```

Comment:
consider only using on stack component and just passing in a variable in a constructor. That way you can create two instance of the same class. 

Also, do we need all three files for each component? can we simplify by combining?

Reply:
We could; I think it might be a bit confusing though. The point is these are supposed to represent different parts of the app (like 'Events', 'Users', etc...). So in this example we could definitely simplify it, but that's not what it would look like in an actual app.